srk/src/dD.ml

```diff
@@ -175,6 +177,155 @@ let minimal_faces polyhedron : (V.t * ((constraint_kind * V.t) list)) list =
       List.iter log_face defining;
       defining)
 
+module IntegerHull = struct
+
+  module D = Linear.MakeDenseConversion(SrkUtil.Int)(V)
+
+  let densify ctx v =
+    let array = Array.make (D.dim ctx) (ZZ.mpz_of ZZ.zero) in
+    BatEnum.iter
+      (fun (a, i) ->
+        array.(D.int_of_dim ctx i) <- ZZ.mpz_of (Option.get (QQ.to_zz a)))
+      (V.enum v);
+    Array.to_list array
+
+  let sparsify ctx v =
+    BatList.fold_lefti (fun vec i a ->
+        V.set (D.dim_of_int ctx i) (QQ.of_zz (ZZ.of_mpz a)) vec)
+      V.zero
+      v
+
+  (* TODO: This is basically Cone's [hilbert_basis] function, but there is
+     a dependency cycle between Cone and DD.
+   *)
+  let hilbert_basis lines rays =
```

Comment:
Leave the dense variant here and put the sparse one in cone.ml?

Reply:
The one here is also handling sparse vectors because that's what we get from within DD. Do you mean to export this in DD.mli with the signature [lines:QQVector.t list -> rays:QQVector.t list -> QQVector.t list] and have the one in cone.ml have its current signature [t -> QQVector.t list] call out to this one?

Reply:
Within DD, vectors are dense (but yes, represented by the type QQVector.t).  I think the interface (lines:QQVector.t list -> rays:QQVector.t list -> QQVector.t list) or (rays:QQVector.t list -> QQVector.t list) in DD.ml is reasonable.